File: theseus_gui/src-tauri/src/api/metadata.rs
use crate::api::Result;
use daedalus::minecraft::VersionManifest;
use daedalus::modded::Manifest;

/// Gets the game versions from daedalus
#[tauri::command]
pub async fn metadata_get_game_versions() -> Result<VersionManifest> {
    Ok(theseus::metadata::get_minecraft_versions().await?)
}

/// Gets the fabric versions from daedalus
#[tauri::command]
pub async fn metadata_get_fabric_versions() -> Result<Manifest> {
    Ok(theseus::metadata::get_fabric_versions().await?)
}

/// Gets the forge versions from daedalus
#[tauri::command]
pub async fn metadata_get_forge_versions() -> Result<Manifest> {
    Ok(theseus::metadata::get_forge_versions().await?)
}

/// Gets the forge versions from daedalus
#[tauri::command]
pub async fn metadata_get_quilt_versions() -> Result<Manifest> {
    Ok(theseus::metadata::get_quilt_versions().await?)
}

File: apps/theseus_gui/src/helpers/metadata.js
import { invoke } from '@tauri-apps/api/tauri'

/// Gets the game versions from daedalus
// Returns a VersionManifest
export async function get_game_versions() {
  return await invoke('plugin:metadata|metadata_get_game_versions')
}

// Gets the fabric versions from daedalus
// Returns Manifest
export async function get_fabric_versions() {
  const c = await invoke('plugin:metadata|metadata_get_fabric_versions')
  console.log('Getting fabric versions', c)
  return c
}

// Gets the forge versions from daedalus
// Returns Manifest
export async function get_forge_versions() {
  const c = await invoke('plugin:metadata|metadata_get_forge_versions')
  console.log('Getting forge versions', c)
  return c
}

// Gets the quilt versions from daedalus
// Returns Manifest
export async function get_quilt_versions() {
  const c = await invoke('plugin:metadata|metadata_get_quilt_versions')
  console.log('Getting quilt versions', c)
  return c
}

// Gets the neoforge versions from daedalus
// Returns Manifest
export async function get_neoforge_versions() {
  const c = await invoke('plugin:metadata|metadata_get_neoforge_versions')
  console.log('Getting neoforge versions', c)
  return c
}
